### PR TITLE
pin pytest

### PR DESCRIPTION
**Pull request by nabobalis**

### CODE REVIEW

**Comment by Cadair** [comment]
Presumably related to https://github.com/astropy/pytest-doctestplus/pull/119 ?

**Comment by nabobalis** [comment]
Yes.

File: sunpy/tests/figure_hashes_mpl_321_ft_261_astropy_401post1.json [hunk 1]
```diff
@@ -24,19 +24,18 @@
     "sunpy.map.tests.test_plotting.test_plot_rotated_aia171": "4c96dbf9b4374283b9669c713a95fbd0cd654dfc75d293dd0c99073ce1a0039e",
```
Comment (Cadair) [inline sunpy/tests/figure_hashes_mpl_321_ft_261_astropy_401post1.json:24]: why the changes to the hashes? Including one going missing?!

Comment (nabobalis) [reply sunpy/tests/figure_hashes_mpl_321_ft_261_astropy_401post1.json:24]: That's what the CI reports.

**Comment by nabobalis** [comment]
Compiling cryptography is failing on the 32bit build. I think but I am not 100% sure, the version of openssl on that version of manylinux2010 is too old. 

Their docs say:

> We test compiling with clang as well as gcc and use the following OpenSSL releases:
> 
> OpenSSL 1.0.2-latest
> OpenSSL 1.1.0-latest
> OpenSSL 1.1.1-latest

The version on manylinux2010 is 1.0.1e-58. We could try bumping the 32bit image to 2014 to see if that fixes it?

**Comment by nabobalis** [comment]
Dev build is failing due to:

```python
    aiamap.plot()
  File "/home/vsts/work/1/s/.tox/py38-devdeps/lib/python3.8/site-packages/astropy/units/decorators.py", line 251, in wrapper
    return_ = wrapped_function(*func_args, **func_kwargs)
  File "/home/vsts/work/1/s/.tox/py38-devdeps/lib/python3.8/site-packages/sunpy/map/mapbase.py", line 2052, in plot
    plot_settings = copy.deepcopy(self.plot_settings)
  File "/opt/hostedtoolcache/Python/3.8.3/x64/lib/python3.8/copy.py", line 146, in deepcopy
    y = copier(x, memo)
  File "/opt/hostedtoolcache/Python/3.8.3/x64/lib/python3.8/copy.py", line 230, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/opt/hostedtoolcache/Python/3.8.3/x64/lib/python3.8/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/opt/hostedtoolcache/Python/3.8.3/x64/lib/python3.8/copy.py", line 270, in _reconstruct
    state = deepcopy(state, memo)
  File "/opt/hostedtoolcache/Python/3.8.3/x64/lib/python3.8/copy.py", line 146, in deepcopy
    y = copier(x, memo)
  File "/opt/hostedtoolcache/Python/3.8.3/x64/lib/python3.8/copy.py", line 230, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/opt/hostedtoolcache/Python/3.8.3/x64/lib/python3.8/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/opt/hostedtoolcache/Python/3.8.3/x64/lib/python3.8/copy.py", line 270, in _reconstruct
    state = deepcopy(state, memo)
  File "/opt/hostedtoolcache/Python/3.8.3/x64/lib/python3.8/copy.py", line 146, in deepcopy
    y = copier(x, memo)
  File "/opt/hostedtoolcache/Python/3.8.3/x64/lib/python3.8/copy.py", line 230, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/opt/hostedtoolcache/Python/3.8.3/x64/lib/python3.8/copy.py", line 153, in deepcopy
    y = copier(memo)
  File "/home/vsts/work/1/s/.tox/py38-devdeps/lib/python3.8/site-packages/matplotlib/transforms.py", line 143, in __copy__
    raise NotImplementedError(
NotImplementedError: TransformNode instances can not be copied. Consider using frozen() instead.
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -54,6 +54,9 @@ jobs:
         - openssl-devel
         - freetype-devel
         - libpng-devel
+        - libffi-devel
+        - python-devel
```
Comment (dstansby) [inline azure-pipelines.yml:58]: Why python-devel?

Comment (nabobalis) [reply azure-pipelines.yml:58]: It's what cryptography said to pre-install in order to compile the library.

Comment (dstansby) [reply azure-pipelines.yml:58]: Could you stick a comment above in the file saying that's why they're here? And is it possible to install just for 32bit instead of all jobs?

Comment (nabobalis) [reply azure-pipelines.yml:58]: It is only installed for the 32bit builds.

Comment (nabobalis) [inline azure-pipelines.yml:58]:
```suggestion
```
